include/pybind11/operators.h

```diff
@@ -103,6 +105,7 @@ template <typename B, typename L, typename R> struct op_impl<op_##id, op_l, B, L
     static char const* name() { return "__" #id "__"; }                                \
     static auto execute(L &l, const R &r) -> decltype(expr) { return expr; }           \
     static B execute_cast(L &l, const R &r) { return B(expr); }                        \
+    static return_value_policy rvp() { return return_value_policy::reference; }        \
```

Comment:
If there is a way to check the return type of `execute` to narrow the application of `rvp::reference`, please advise. I.e., similar to `decltype(expr)` but without the arguments. However, even in cases where the return value is not a reference, this will still invoke copy construction, so this is more of a self-documentating-code problem.